### PR TITLE
fix dimension anomalies bug

### PR DESCRIPTION
**Pull request by IDoneShaveIt**
1. Fixed dimension anomalies bug
2. Fixed e2e test for dimension anomalies
3. Fixed inconsistency in the way we save models and tags

### CODE REVIEW

File: macros/edr/dbt_artifacts/upload_dbt_models.sql [hunk 1]
```diff
@@ -36,8 +36,16 @@
     {% set config_meta_dict = elementary.safe_get_with_default(config_dict, 'meta', {}) %}
     {% set meta_dict = elementary.safe_get_with_default(node_dict, 'meta', {}) %}
     {% do meta_dict.update(config_meta_dict) %}
-    {% set owner = meta_dict.get('owner') %}
-
+    {% set formatted_owner = [] %}
```
Comment (IDoneShaveIt) [inline macros/edr/dbt_artifacts/upload_dbt_models.sql:39]: Owners now will be saved as list (even when given only single owner).
This allows us to handle cases when the users provide owners like:
1. "owner1"
2. "owner1, owner2"
3. ["owner1"]
4. ["owner1", "owner2"]

File: macros/edr/data_monitoring/anomaly_detection/get_anomaly_scores_query.sql [hunk 2]
```diff
@@ -105,11 +105,11 @@
...
-                stddev(metric_value) over (partition by metric_name, full_table_name, column_name order by edr_daily_bucket asc rows between {{ elementary.get_config_var('days_back') }} preceding and current row) as training_stddev,
-                count(metric_value) over (partition by metric_name, full_table_name, column_name order by edr_daily_bucket asc rows between {{ elementary.get_config_var('days_back') }} preceding and current row) as training_set_size,
-                last_value(bucket_end) over (partition by metric_name, full_table_name, column_name order by edr_daily_bucket asc rows between {{ elementary.get_config_var('days_back') }} preceding and current row) training_end,
-                first_value(bucket_end) over (partition by metric_name, full_table_name, column_name order by edr_daily_bucket asc rows between {{ elementary.get_config_var('days_back') }} preceding and current row) as training_start
+                avg(metric_value) over (partition by metric_name, full_table_name, column_name, dimension, dimension_value order by edr_daily_bucket asc rows between {{ elementary.get_config_var('days_back') }} preceding and current row) as training_avg,
```
[the remark this answers is not in the document]
Comment (elongl) [reply macros/edr/data_monitoring/anomaly_detection/get_anomaly_scores_query.sql:108]: Could you explain how come the test worked so far without this?

[the remark this answers is not in the document]
Comment (IDoneShaveIt) [reply macros/edr/data_monitoring/anomaly_detection/get_anomaly_scores_query.sql:108]: Becaus of the group by it still got different values for each dimension, but it got wrong values.
When the anoaly differential was big enough it did "found" anomalies but it was statisticly.
The data we had in our e2e was a big gap on the volume so it did pass, and this is why we thought that it was working

File: integration_tests/models/schema.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ version: 2
 models:
   - name: any_type_column_anomalies
     meta:
-      owner: [ '@edr', 'ele@data.com, another@mail.com' ]
+      owner: ["@edr", "ele@data.com", "another@mail.com"]
```
[the remark this answers is not in the document]
Comment (IDoneShaveIt) [reply integration_tests/models/schema.yml:6]: Format on save 😱

**Pull request merged**